File: CS/Form1.cs
using DevExpress.Utils;
using DevExpress.XtraBars;
using System;
using System.Linq;

namespace dxSample {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        Random r = new Random();

        public Form1() {
            InitializeComponent();
        }

        private void barEditItem1_EditValueChanged(object sender, EventArgs e) {
            var editItem = sender as BarEditItem;
            var value = Convert.ToInt32(editItem.EditValue);
            PointFloat[] points = new PointFloat[value];

            for (int i = 0; i < value; i++) {
                points[i] = new PointFloat() { X = Convert.ToSingle(r.NextDouble()), Y = r.Next(0, 2) == 0 ? 0 : 1 };
            }

            diagramShape1.ConnectionPoints = new DevExpress.XtraDiagram.PointCollection(points);
        }
    }
}

File: VB/Form1.vb
Imports DevExpress.Utils
Imports DevExpress.XtraBars
Imports System

Namespace dxSample

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Private r As Random = New Random()

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub barEditItem1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim editItem = TryCast(sender, BarEditItem)
            Dim value = Convert.ToInt32(editItem.EditValue)
            Dim points As PointFloat() = New PointFloat(value - 1) {}
            For i As Integer = 0 To value - 1
                points(i) = New PointFloat() With {.X = Convert.ToSingle(r.NextDouble()), .Y = If(r.Next(0, 2) = 0, 0, 1)}
            Next

            diagramShape1.ConnectionPoints = New DevExpress.XtraDiagram.PointCollection(points)
        End Sub
    End Class
End Namespace
